### DISCUSSION TITLE
Show animation of the change in a strength of a habit (tick/untick)

### DISCUSSION THREAD

**Discussion by mikkovedru** [iSoron/uhabits#169] [Feature Requests]
- Current situation: This app is all about good habit reinforcement. Now it displays a percentage score, which tells us the strength of a habit. Newer check-marks count for more than the older ones.
- Problem: Each time I go to app to tick a check mark on "Eat healthy" I want to feel proud. And I do. I can also see the percentage growing towards the full circle. Unfortunately the change isn't that big. So now I have to do a three-step process every time I am about to complete a habit:
  a) go inside a habit and see the current percentage
  b) go out and complete the habit for today
  c) go inside a habit, see the new percentage and calculate the difference
  This is pretty tedious...
- Solution: Every time there is a habit being ticked, the interface should show an animation of a colored (green? the color of the habit?) of the number ("+3.4%"). Maybe even with exclamation points, happy face and cool sound effect. Obviously opposite for unticking the habit.

I know that this is a small thing. But small things matter. And I can see from the great minimalistic interface the app currently has, that developer also thinks that small things like that are important.

I think this feature would improve the positive reinforcement of ticking the habit. Habit formation books talk about the habit loop of "Cue -> Routine -> Reward" and making the whole loop stronger. Talk about the strength of the Reward, how important it is. The App is such a reward. And since people have different preferences (audio, visual, etc), it would be efficient to use as many ways to increase the enjoyment/reward in the app as possible.

Thanks in advance for considering.

**Comment by iSoron** [comment]
Sorry about the delay. I think this is an interesting feature, although I am not sure how should the animation look like. Mockups are very welcome. The easiest way would be to show this information in the snackbar, but I'm sure it can be done in a better way.

**Comment by deutrino** [comment]
I came here to beg for a similar but simpler feature on the same behavioral basis. I would like a small, fun, unobtrusive animation when I hit the check mark on a habit! It's positive reinforcement.

I had no other ideas about information being contained therein (the "+3.4%" idea). That probably requires more thought, especially around the potential to become less motivating as a habit is stronger. But a simple animation could perhaps be implemented sooner?

I made a video of two apps I use which have an animation for "liking" posts. I think something similar for the check mark (but no animation for unchecking) would be nice. 

I also think that if you go to the trouble to add animated check marks, perhaps allowing the user to select from one of, say, three animation styles (and/or disable them entirely) would be a good idea. I say this because of the fact that we want positive reinforcement - for some users, positive reinforcement might look like some little hearts and stars, where another would find that repulsive and prefer a simple green checkmark to float upwards, or prefer nothing at all. 

Here is the video of how Twidere and Tumblr animate a UI icon:

https://mega.nz/#!ghckSTTJ!duiawxaSm-9RdLQxSsgL6KGWm4TF_qhWwCFwuekshII

Thank you for writing this app, it is by far and away the best habit encouragement app I've ever used. Nothing else even comes close to how effective this is. 😊

(I apologize for the MEGA link, it will bug you to install an app but there is a link available to download in the browser. The other service I use expires files after 60 days.)

**Comment by ke352802081770314** [comment]
I think this functionality would be very pleasing. 
It would already be good enough to me if it was displayed near the bottom like now, when I short-click (I have enabled "long press" to toggle/input) the "press and hold to enter value" message is displayed (sorry I am not familiar with the terms in smartphone UI, so e.g. I haven't a clue what a snackbar is),  displaying e.g. "+2.3% (55.0% -> 57.3%)" <- note if you only display numbers  you don't need language versions ;)
